### PR TITLE
Re-Former

### PR DESCRIPTION
**Pull request by ZeenatLawal**
In this project we:

- Created a Re-former project where a user can create a form.
- Built the user model and added validations to the inputs.
- Used the form helper tags to build the form.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,47 @@
...
+### To get a local copy run the following steps:
 
-* Database creation
+- [Copy this link](git@github.com:Mhdez221993/re-former.git)
+- Open your terminal or command line
+- Run `git clone` and Paste the link
+- Open the folder with your code editor
+- Run `bundle install` to install all dependecies
+- Run `rails db:migrate` to create the database tables
+- Run `rails console` to test the project
```
Comment (enaburekhan) [inline README.md]: The detailed instructions for running the app are currently incomplete.  Let me add some instructions. Continuing from where you stopped:
Before `rails console`. First I need to test if the app is working as expected using 

- `rails server`
- `http://localhost:3000/users/new` to create a new user
- `http://localhost:3000/users/1/edit` to edit an existing user

To confirm this:

- Run `rails console`
- `User.all` to display the exiting user.

You can refactor the above instructions to produce something similar
Lastly, I couldn't run the app without running `yarn install`. So do well to include it in the instructions

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (enaburekhan) [inline app/controllers/users_controller.rb]: Remove commented codes. They make the codebase to be untidy, difficult to read and understand. Use git history if you must view previous codes.

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(@user.errors.count, "error") %> prohibited this user from being saved:</h2>
+
+      <ul>
+      <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (enaburekhan) [inline app/views/users/edit.html.erb]: **Optional:**

- [ ] Kindly move the logic to a helper. There should be absolutely NO logic in views. 
Views are only for displaying information that became available to it from its corresponding controller not generating it. Check this [article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) on how to use a helper in rails.

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      <div id="error_explanation">
+        <h2><%= pluralize(@user.errors.count, "error") %> prohibited this user from being saved:</h2>
+
+        <ul>
+        <% @user.errors.full_messages.each do |msg| %>
+          <li><%= msg %></li>
+        <% end %>
+        </ul>
+      </div>
+    <% end %>
```
Comment (enaburekhan) [inline app/views/users/new.html.erb]: **Optional:**

- [ ] Kindly also move the logic to a helper. The reason, as already highlighted above.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi team!
Great job on this project:clap::clap:.**

**Major highlights**

- The PR and commit messages are very descriptive and professional
- The project strictly follows the MVC pattern.

However, we still need to make a few changes to get the project approved:muscle:.

**To improve**

- [x] View

- Add the commented out form in the new.html.erb. Kindly leave it in the commented form. 

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_with.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

- Also, add the commented out form_tag to the new.html.erb. Also leave them in the commented out form. Reason, already given above.

- [x] Check my inline comments below and make the required changes.

**Optional**

- [x] It will be nice to make the changes tag as optional in the inline comments below.

- [x] form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

Make the above-required changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: 
------------------
Hello @ZeenatLawal @Mhdez221993 great job working on the required changes :clap: your project is working really nice. :100: 
-----------------
Now you can merge this branch :smile: 
HAPPY CODING!!!! AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover:

**Pull request merged**